File: client/src/app/_models/UserParams.ts
import {User} from './User';
import {PageParams} from './PageParams';

export class UserParams extends PageParams {

  maxAge: number = 99;
  minAge: number = 18;
  gender: string;
  orderBy: string = "lastActive"

  constructor(user: User) {
    super();
    this.gender = user.gender === 'male' ? 'female' : 'male';
  }
}

File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string Gender { get; set; }
        public string CurrentUserName { get; set; }
        public int MaxAge { get; set; } = 150;
        public int MinAge { get; set; } = 18;
        public string OrderBy { get; set; } = "lastActive";


    }
}